base/actions/actions.py

```diff
@@ -837,8 +829,25 @@ async def submit(
                     if i == config.HTTP_RETRIES - 1:
                         raise e
         self.send_risk_to_user(dispatcher, risk, tracker)
-        self.send_study_a_message(dispatcher, study_a_arm, tracker)
         self.send_post_risk_prompts(dispatcher, risk, tracker)
+
+        return [SlotSet("study_a_arm", study_a_arm)]
+
+
+class ActionSendStudyMessages(Action):
+    def name(self) -> Text:
+        return "action_send_study_messages"
+
+    def run(
+        self,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> List[Dict[Text, Any]]:
+        study_a_arm = tracker.get_slot("study_a_arm")
+        if study_a_arm and study_a_arm != "C":
+            sleep(config.STUDY_A_MESSAGE_DELAY)
```

Comment:
The Rasa actions server uses the Sanic webserver library, which is async. By doing a normal `sleep`, you'll be blocking the webserver for the duration of the sleep, and it won't be able to process any new requests.

A better way would be to define the function as `async def run` (Rasa SDK allows `run` to be either sync or async), and then use `await asyncio.sleep(...)` for the sleep. That way the webserver will be able to server other requests while the sleep is happening.